auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java

```diff
@@ -51,16 +51,16 @@ public final void filter(ContainerRequestContext requestContext) {
             return;
         }
 
+        UnverifiedJsonWebToken jwt;
         try {
-            UnverifiedJsonWebToken jwt = UnverifiedJsonWebToken.of(
-                    AuthHeader.valueOf(rawAuthHeader).getBearerToken());
-
-            setUnverifiedContext(requestContext, USER_ID_KEY, jwt.getUnverifiedUserId());
-            jwt.getUnverifiedSessionId().ifPresent(s -> setUnverifiedContext(requestContext, SESSION_ID_KEY, s));
-            jwt.getUnverifiedTokenId().ifPresent(s -> setUnverifiedContext(requestContext, TOKEN_ID_KEY, s));
-        } catch (Throwable t) {
-            log.debug("Unable to process auth header.", t);
+            jwt = UnverifiedJsonWebToken.of(AuthHeader.valueOf(rawAuthHeader).getBearerToken());
+        } catch (IllegalArgumentException e) {
```

Comment:
is this the only possible reason that this code can throw? if not, other reasons will prevent us from serving _any_ request

Reply:
Hmm you've got a point. And these are all in the same project, though `UnverifiedJsonWebToken` is in a different module, so I guess the error I caught was due to bad version forces.

Reply:
However.. it might be preferrable to fail fast here, because if the tried code throws anything else that we didn't expect, services (and tests) should know about it immediately.

Reply:
It’s not okay for this for-logging-purposes filter to halt requests. Errors are logged and provide plenty of signal for debug.